dart/lib/src/sentry_options.dart

```diff
@@ -139,8 +148,9 @@ class SentryOptions {
     this.environmentAttributes,
     this.compressPayload,
     this.httpClient,
+    Transport transport,
     ClockProvider clock = getUtcDateTime,
-  }) {
+  }) : _transport = transport ?? NoOpTransport() {
```

Comment:
why do we have this here? its already `NoOpTransport` by default, we expose a setter already

Reply:
we can't access the setter from the constructor, so if we don't want a null transport we needs this.

Reply:
but the field declaration already has NoOpTransport as a value, it should not belong to the ctor, a setter wont be able to set it to null either, because on the setter we need to check if its null and not allow to set it

Reply:
we can't write this : 
```dart
SentryOptions({
    this.dsn,
    this.environmentAttributes,
    this.compressPayload,
    this.httpClient,
    this.transport,
    ClockProvider clock = getUtcDateTime,
  })
```
cause this.transport refers to setter and it can't be accessed from here

Reply:
But we dont need a transport in the ctor, as i said, we dont need any ctor params, options is a mutable class and all fields have setters, you could even remove this ctor altogether

Reply:
ah sorry. in french we say : "I understand fast, but you have to explain me long time" :D

Reply:
✅ removed !